config/reader.go

```diff
@@ -437,6 +438,36 @@ func processConfigLocalConfig(unprocessedConfig *Config, logger logging.Logger)
 	return processConfig(unprocessedConfig, false, logger)
 }
 
+// additionalModuleEnvVars will get additional environment variables for modules using other parts of the config.
+func additionalModuleEnvVars(cloud *Cloud, auth AuthConfig) map[string]string {
+	env := make(map[string]string)
+	if cloud != nil {
+		env[rutils.PrimaryOrgIDEnvVar] = cloud.PrimaryOrgID
+		env[rutils.LocationIDEnvVar] = cloud.LocationID
+		env[rutils.MachineIDEnvVar] = cloud.MachineID
+		env[rutils.MachinePartIDEnvVar] = cloud.ID
+	}
+	for _, handler := range auth.Handlers {
+		if handler.Type != rpc.CredentialsTypeAPIKey {
+			continue
+		}
+		apiKeys := ParseAPIKeys(handler)
+		if len(apiKeys) == 0 {
+			continue
+		}
+		// the keys come in unsorted, so sort the keys so we'll always get the same API key
+		// if there are no changes
```

Comment:
How do we know that the first API key is the one the module wants?

Reply:
we don't - it was mentioned during 6 week check-in that we should just make it easy to access, so opted for this instead of returning all of them.